prow/plugins/approve/approve.go

```diff
@@ -171,6 +172,63 @@ func handleGenericComment(log *logrus.Entry, ghc githubClient, repo approvers.Re
 	)
 }
 
+// handleReviewEvent should only handle reviews that have no approval command.
+// Reviews with approval commands will be handled by handleGenericCommentEvent.
+func handleReviewEvent(pc plugins.PluginClient, re github.ReviewEvent) error {
+	botName, err := pc.GitHubClient.BotName()
+	if err != nil {
+		return err
+	}
+
+	opts := optionsForRepo(pc.PluginConfig, re.Repo.Owner.Login, re.Repo.Name)
+
+	// Check for an approval command is in the body. If one exists, let the
+	// genericCommentEventHandler handle this event. Approval commands override
+	// review state.
+	if approvalCommandMatcher(botName, opts.LgtmActsAsApprove, opts.ReviewActsAsApprove)(&comment{Body: re.Review.Body, Author: re.Review.User.Login}) {
+		return nil
+	}
+
+	// Check for an approval command via review state. If none exists, don't
+	// handle this event.
+	if !approvalCommandMatcher(botName, opts.LgtmActsAsApprove, opts.ReviewActsAsApprove)(&comment{Author: re.Review.User.Login, ReviewState: re.Review.State}) {
```

Comment:
nit: I think it would be a little cleaner to split the `approvalCommandMatcher` func into two. One that checks for the command and one that checks for review state. Then we can call the appropriate functions here instead of calling the same function twice with different args.
Existing uses of `approvalCommandMatcher` that require both checks could use a wrapper func that calls both check functions.

Reply:
+1

Reply:
I split this into two checks: `isApprovalCommand` and `isApprovalState`. Those functions are used directly by the event handlers to do pre-handle filtering. A new function `approvalMatcher` returns a function that ORs the results of those checks together for use with `filterComments` in `handle()`.